Add entity state formatting documentation and blog article (#1891)
home-assistant/developers.home-assistant: docs/frontend/data.md

sidebar_label: "Data"
---

> The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provide helper for formatting entity state.

Reviewer: ```suggestion
The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provides a helper for formatting entity state.
```

> The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provides helpers for formatting entity state.

Author: ```suggestion
The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provides helpers to format entity state.
```

> The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provides a helper for formatting entity state.

Author: ```suggestion
The frontend passes a single `hass` object around. This object contains the latest state, allows you to send commands back to the server and provides helpers for formatting entity state.
```

:::info
We're moving away from API calls and are migrating everything to `hass.callWS(message)` calls.
:::

## Entity state formatting

> These methods allows you to format the state and attributes of an entity. The value will be localized using user profile settings (language, number format, date format, timezone) and unit of measurement.

Author: ```suggestion
These methods allow you to format the state and attributes of an entity. The value will be localized using user profile settings (language, number format, date format, timezone) and unit of measurement.
```

### `hass.formatEntityState(stateObj, state)`

Format the state of an entity. You need to pass the entity state object.

```js
hass.formatEntityState(hass.states["light.my_light"]); // "On"
```

You can force the state value using the second optional parameter.

```js
hass.formatEntityState(hass.states["light.my_light"], 'off'); // "Off"
```

### `hass.formatEntityAttributeValue(stateObj, attribute, value)`

Format the attribute value of an entity. You need to pass the entity state object and the attribute name.

```js
hass.formatEntityAttributeValue(hass.states["climate.thermostat"], "current_temperature"); // "20.5 °C"
```

> You can force the state value using the second optional parameter.

Reviewer: ```suggestion
You can force the state value using the third optional parameter.
```

```js
hass.formatEntityAttributeValue(hass.states["climate.thermostat"], "current_temperature", 18); // "18 °C"
```

### `hass.formatEntityAttributeName(stateObj, attribute)`

Format the attribute name of an entity. You need to pass the entity state object and the attribute name.

```js

> hass.formatEntityAttributeName(hass.states["climate.thermostat", "current_temperature"], ); // "Current temperature"

Reviewer: Is this syntax correct, ie including the attribute name inside the brackets?

Author: No, it is not. Thanks!

> hass.formatEntityAttributeName(hass.states["climate.thermostat", "current_temperature"], ); // "Current temperature"

Author: ```suggestion
hass.formatEntityAttributeName(hass.states["climate.thermostat"], "current_temperature"); // "Current temperature"
```
